src/core/utils/intersectUtils.js

```diff
@@ -1,10 +1,74 @@
-import { Box3 } from 'three';
-import { arrayToBox } from '../../utils/ArrayBoxUtilities.js';
+export function intersectRay( nodeIndex32, array, ray ) {
 
-const _boundingBox = /* @__PURE__ */ new Box3();
-export function intersectRay( nodeIndex32, array, ray, target ) {
+	let tmin, tmax, tymin, tymax, tzmin, tzmax;
 
-	arrayToBox( nodeIndex32, array, _boundingBox );
-	return ray.intersectBox( _boundingBox, target );
+	const invdirx = 1 / ray.direction.x,
+		invdiry = 1 / ray.direction.y,
+		invdirz = 1 / ray.direction.z;
+
+	const ox = ray.origin.x;
+	const oy = ray.origin.y;
+	const oz = ray.origin.z;
+
+	let minx = array[ nodeIndex32 ];
+	let maxx = array[ nodeIndex32 + 3 ];
+
+	let miny = array[ nodeIndex32 + 1 ];
+	let maxy = array[ nodeIndex32 + 3 + 1 ];
+
+	let minz = array[ nodeIndex32 + 2 ];
+	let maxz = array[ nodeIndex32 + 3 + 2 ];
+
+	if ( invdirx >= 0 ) {
+
+		tmin = ( minx - ox ) * invdirx;
+		tmax = ( maxx - ox ) * invdirx;
+
+	} else {
+
+		tmin = ( maxx - ox ) * invdirx;
+		tmax = ( minx - ox ) * invdirx;
+
+	}
+
+	if ( invdiry >= 0 ) {
+
+		tymin = ( miny - oy ) * invdiry;
+		tymax = ( maxy - oy ) * invdiry;
+
+	} else {
+
+		tymin = ( maxy - oy ) * invdiry;
+		tymax = ( miny - oy ) * invdiry;
+
+	}
+
+	if ( ( tmin > tymax ) || ( tymin > tmax ) ) return null;
+
+	if ( tymin > tmin || isNaN( tmin ) ) tmin = tymin;
+
+	if ( tymax < tmax || isNaN( tmax ) ) tmax = tymax;
+
+	if ( invdirz >= 0 ) {
+
+		tzmin = ( minz - oz ) * invdirz;
+		tzmax = ( maxz - oz ) * invdirz;
+
+	} else {
+
+		tzmin = ( maxz - oz ) * invdirz;
+		tzmax = ( minz - oz ) * invdirz;
+
+	}
+
+	if ( ( tmin > tzmax ) || ( tzmin > tmax ) ) return null;
+
+	if ( tzmax < tmax || tmax !== tmax ) tmax = tzmax;
```

Comment:
It looks like a line here has been removed? I understand it might be useless but can we keep it so the function matches three.js' more exactly incase we need to update it later. If we decide to add a distance output later (as you have suggested for raycaster.near and far) it will also be easy to forget this line has been removed and cause bugs.

Reply:
I restored it but DeepScan gives error. I leave it commented.